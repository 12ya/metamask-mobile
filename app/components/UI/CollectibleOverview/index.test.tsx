import React from 'react';
import CollectibleOverview from './';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { backgroundState } from '../../../util/test/initial-root-state';

const mockStore = configureMockStore();
const initialState = {
  collectibles: {
    favorites: {},
  },
  engine: {
    backgroundState,
  },
};
const store = mockStore(initialState);

jest.mock('../../../core/Engine', () => ({
  context: {
    NetworkController: {
      getNetworkClientById: () => ({
        configuration: {
          rpcUrl: 'https://mainnet.infura.io/v3',
          chainId: '0x1',
          ticker: 'ETH',
          nickname: 'Ethereum mainnet',
          rpcPrefs: {
            blockExplorerUrl: 'https://etherscan.com',
          },
        },
      }),
      state: {
        networkConfigurations: {
          '673a4523-3c49-47cd-8d48-68dfc8a47a9c': {
            id: '673a4523-3c49-47cd-8d48-68dfc8a47a9c',
            rpcUrl: 'https://mainnet.infura.io/v3',
            chainId: '0x1',
            ticker: 'ETH',
            nickname: 'Ethereum mainnet',
            rpcPrefs: {
              blockExplorerUrl: 'https://etherscan.com',
            },
          },
        },
        selectedNetworkClientId: '673a4523-3c49-47cd-8d48-68dfc8a47a9c',
        networkMetadata: {},
      },
    },
  },
}));

describe('CollectibleOverview', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <CollectibleOverview
          collectible={{
            name: 'Leopard',
            tokenId: 6904,
            address: '0x06012c8cf97BEaD5deAe237070F9587f8E7A266d',
            externalLink: 'https://nft.example.com',
            tradable: true,
          }}
        />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
