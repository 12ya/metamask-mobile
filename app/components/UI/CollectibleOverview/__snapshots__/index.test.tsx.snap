// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectibleOverview should render correctly 1`] = `
<View
  collapsable={false}
  forwardedRef={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    {
      "backgroundColor": "#ffffff",
      "borderTopEndRadius": 8,
      "borderTopStartRadius": 8,
      "flex": 0,
      "transform": [
        {
          "translateY": 0,
        },
      ],
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingVertical": 16,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#bbc0c5",
          "borderRadius": 4,
          "height": 5,
          "width": 48,
        }
      }
    />
  </View>
  <RCTSafeAreaView>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          {
            "paddingHorizontal": 16,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            [
              {
                "color": "#141618",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 30,
                "fontWeight": "400",
                "marginVertical": 2,
              },
              undefined,
              undefined,
              {
                "fontFamily": "EuclidCircularB-Bold",
                "fontWeight": "600",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#141618",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "marginVertical": 0,
              },
              {
                "fontSize": 24,
                "marginBottom": 3,
              },
            ]
          }
        >
          Leopard
        </Text>
        <Text
          style={
            [
              {
                "color": "#141618",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 30,
                "fontWeight": "400",
                "marginVertical": 2,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#141618",
              },
              undefined,
              undefined,
              undefined,
              {
                "fontSize": 16,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "marginVertical": 0,
              },
              undefined,
            ]
          }
        >
          #
          6904
        </Text>
      </View>
      <View
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            {
              "flexDirection": "row",
              "marginTop": 16,
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityRole="button"
          accessible={true}
          activeOpacity={0.2}
          onPressOut={[Function]}
          style={
            [
              [
                {
                  "borderRadius": 100,
                  "justifyContent": "center",
                  "padding": 15,
                },
                {
                  "backgroundColor": "#ffffff",
                  "borderColor": "#0376c9",
                  "borderWidth": 1,
                  "padding": 8,
                },
                [
                  {
                    "alignItems": "center",
                    "borderWidth": 1.5,
                    "justifyContent": "center",
                  },
                  {
                    "height": 54,
                    "width": 54,
                  },
                  {
                    "marginRight": 16,
                  },
                ],
              ],
              null,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#141618",
                  "fontFamily": "EuclidCircularB-Regular",
                  "fontSize": 30,
                  "fontWeight": "400",
                  "marginVertical": 2,
                },
                undefined,
                undefined,
                {
                  "fontFamily": "EuclidCircularB-Bold",
                  "fontWeight": "600",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "color": "#0376c9",
                },
                undefined,
                undefined,
                {
                  "marginVertical": 0,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 32,
                  },
                  undefined,
                  {
                    "fontFamily": "EvilIcons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          accessibilityRole="button"
          accessible={true}
          activeOpacity={0.2}
          onPressOut={[Function]}
          style={
            [
              [
                {
                  "borderRadius": 100,
                  "justifyContent": "center",
                  "padding": 15,
                },
                {
                  "backgroundColor": "#ffffff",
                  "borderColor": "#0376c9",
                  "borderWidth": 1,
                  "padding": 8,
                },
                [
                  {
                    "alignItems": "center",
                    "borderWidth": 1.5,
                    "justifyContent": "center",
                  },
                  {
                    "height": 54,
                    "width": 54,
                  },
                ],
              ],
              null,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#141618",
                  "fontFamily": "EuclidCircularB-Regular",
                  "fontSize": 30,
                  "fontWeight": "400",
                  "marginVertical": 2,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "color": "#0376c9",
                },
                undefined,
                undefined,
                {
                  "marginVertical": 0,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 24,
                  },
                  undefined,
                  {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </Text>
        </TouchableOpacity>
      </View>
    </View>
    <View />
    <View
      style={
        {
          "paddingTop": 24,
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 8,
            "marginHorizontal": 16,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#141618",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 30,
                "fontWeight": "400",
                "marginVertical": 2,
              },
              undefined,
              undefined,
              {
                "fontFamily": "EuclidCircularB-Bold",
                "fontWeight": "600",
              },
              undefined,
              {
                "color": "#141618",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#141618",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "fontSize": 16,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "marginVertical": 0,
              },
              {
                "paddingRight": 10,
              },
            ]
          }
        >
          Asset contract
        </Text>
        <Text
          ellipsizeMode="middle"
          numberOfLines={1}
          onPress={[Function]}
          style={
            [
              {
                "color": "#141618",
                "fontFamily": "EuclidCircularB-Regular",
                "fontSize": 30,
                "fontWeight": "400",
                "marginVertical": 2,
              },
              undefined,
              {
                "textAlign": "right",
              },
              undefined,
              undefined,
              false,
              undefined,
              undefined,
              undefined,
              undefined,
              false,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "fontSize": 16,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#0376c9",
              },
              undefined,
              undefined,
              {
                "marginVertical": 0,
              },
              {
                "flex": 1,
              },
            ]
          }
        >
          0x0601...266d
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
